python/cudf/cudf/core/column/column.py

```diff
@@ -702,7 +702,9 @@ def fillna(
         Returns a copy with null filled.
         """
         return libcudf.replace.replace_nulls(
-            input_col=self, replacement=fill_value, method=method
+            input_col=self.nans_to_nulls(),
```

Comment:
I think you fixed a bug that is resulting in an increase in test pass rate:

![Screenshot 2024-06-05 at 11 43 24 AM](https://github.com/rapidsai/cudf/assets/11664259/48ecde94-c20d-4aa8-9955-7ff198ea3784)

Would you be able to add a test having a mix of nan's & NA's performing a `to_arrow` ? Just wanting to make sure we have it captured in pytest so that we know if it breaks/someone changes this code unintentionally.

Reply:
> Would you be able to add a test having a mix of nan's & NA's performing a to_arrow ?

I assume you meant `fillna` instead of `to_arrow`? If so, I added that test in https://github.com/rapidsai/cudf/pull/15923/commits/0d74275b228bc66f34b3852f9fa5e273e27a5114